ckan/views/resource.py

```diff
@@ -0,0 +1,908 @@
+        u'model': model,
+        u'session': model.Session,
+        u'user': g.user,
+        u'auth_user_obj': g.userobj
+    }
+
+    try:
+        package = get_action(u'package_show')(context, {u'id': id})
+    except (NotFound, NotAuthorized):
+        return base.abort(404, _(u'Dataset not found'))
+
+    try:
+        resource = get_action(u'resource_show')(context, {u'id': resource_id})
+    except (NotFound, NotAuthorized):
+        return base.abort(404, _(u'Resource not found'))
+
+    view = None
+    if request.params.get(u'resource_view', u''):
+        try:
+            view = json.loads(request.params.get(u'resource_view', u''))
+        except ValueError:
+            return base.abort(409, _(u'Bad resource view data'))
+    elif view_id:
+        try:
+            view = get_action(u'resource_view_show')(context, {u'id': view_id})
+        except (NotFound, NotAuthorized):
+            return base.abort(404, _(u'Resource view not found'))
+
+    if not view or not isinstance(view, dict):
+        return base.abort(404, _(u'Resource view not supplied'))
+
+    return h.rendered_resource_view(view, resource, package, embed=True)
+
+
+# FIXME: could anyone think about better name?
+class EditResourceViewView(MethodView):
+    def _prepare(self, id, resource_id):
+        context = {
+            u'model': model,
+            u'session': model.Session,
+            u'user': g.user,
+            u'for_view': True,
+            u'auth_user_obj': g.userobj
+        }
+
+        # update resource should tell us early if the user has privilages.
+        try:
+            check_access(u'resource_update', context, {u'id': resource_id})
+        except NotAuthorized:
+            return bgase.abort(
+                403,
+                _(u'User %r not authorized to edit %s') % (g.user, id)
+            )
+
+        # get resource and package data
+        try:
+            pkg_dict = get_action(u'package_show')(context, {u'id': id})
+            pkg = context[u'package']
+        except (NotFound, NotAuthorized):
+            return base.abort(404, _(u'Dataset not found'))
+        try:
+            resource = get_action(u'resource_show')(
+                context, {
+                    u'id': resource_id
+                }
+            )
+        except (NotFound, NotAuthorized):
+            return base.abort(404, _(u'Resource not found'))
+
+        # TODO: remove
+        g.pkg_dict = pkg_dict
+        g.pkg = pkg
+        g.resource = resource
+
+        extra_vars = dict(
+            data={},
+            errors={},
+            error_summary={},
+            view_type=None,
+            to_preview=False,
+            pkg_dict=pkg_dict,
+            pkg=pkg,
+            resource=resource
+        )
+        return context, extra_vars
+
+    def post(self, package_type, id, resource_id, view_id=None):
+        context, extra_vars = self._prepare(id, resource_id)
+
+        to_preview = request.POST.pop(u'preview', False)
```

Comment:
`request.POST` isn't a thing in flask. This would need to be `request.form` but that's immutable, so we can't call `pop`

Reply:
@smotornyuk can you solve these two issues? we are obviously missing some basic tests here or these would have been picked up